[libgpg-error,libgcrypt,libassuan,gpgme,libgwenhywfar] Update and fix

dg0yt opened the pull request:
Updates ports.
Fixes https://github.com/microsoft/vcpkg/pull/27668#discussion_r1339794415.
Fixes a number of other quirks in particular in the provided config scripts, and for windows hosts (targeting mingw).

Updating libgwenhywfar is out of scope for this PR. (It is not from the gnupg libs familly.) https://github.com/microsoft/vcpkg/issues/31216 remains unfixed. AFAIU libgwenhywfar incorrectly detects and forwards flags from qmake. The fix might be investigated together with an update in another PR.

dg0yt committed:
[libgpg-error] Update to 1.47

dg0yt committed:
[libgcrypt] Update to 1.10.2

dg0yt committed:
[libassuan] Update to 2.5.6

dg0yt commented:
CC @friendlyanon

dg0yt committed:
[libgcrypt] Add fix for getrandom

friendlyanon reviewed (approved):
Builds fine locally. Thanks!

dg0yt commented:
This needs more work for static linkage.

dg0yt committed:
[libgpg-error] Patch gpgrt-config for vcpkg

dg0yt committed:
[libassuan,libgcrypt] Use gpgrt-config

dg0yt committed:
Remove redundant options

jiayuehua committed:
[gpgme] update to 1.22.0

dg0yt committed:
[libgpg-error] Update gpgrt-config patch

dg0yt committed:
[gpgme] Fix and cleanup

dg0yt committed:
[libgwenhywfar] Use gpgrt-config

dg0yt committed:
[libgwenhywfar] Revise portfile

dg0yt committed:
Update versions

dg0yt committed:
[gpgme] Fix pkgconfig fixup

dg0yt committed:
[libgpg-error] Fix gpgrt-config output

The output is burned into downstream config scripts inside quotes.
So we must remove all quotes (added by vcpkg_fixup_config) here.
NB: pkg-config would omit quotes, too.

dg0yt committed:
Fix config fixup for windows hosts

dg0yt committed:
[gpgme] Fake GPG_ERROR_CONFIG to enable gpgme-config

dg0yt committed:
Merge branch 'microsoft:master' into gnupg-libs

BillyONeal commented on ports/libgwenhywfar/portfile.cmake:
```diff
@@ -11,54 +6,61 @@ vcpkg_download_distfile(ARCHIVE
+        "gpg_error_config_args=gpg-error"
+        "LIBGCRYPT_CONFIG=${CURRENT_INSTALLED_DIR}/tools/libgcrypt/bin/libgcrypt-config"
+    OPTIONS_DEBUG
+        "--with-qt5-qmake=${CURRENT_HOST_INSTALLED_DIR}/tools/qt5/debug/bin/qmake"
```
It seems like release build tools should be used even when building debug, as was done before? Or am I missing something in now moc/qmake work?

dg0yt replied:
I can't say for sure, and I didn't find an example. I thought that the debug qmake exists because it is what is needed to find the debug mkspecs... (I really need to fix qt5 cross builds to make these ports usable for me.)
I can revert this part of the change, this qt feature is broken on osx anyways.

BillyONeal replied:
OK, if there's a chance there might be a reason for it rather than it just being blind it's OK. Going to merge as-is since builds passed.

BillyONeal reviewed (changes requested):
This looks really good, only 1 question about qt build tools

BillyONeal commented:
Thanks for all the updates and fixes!
